tutorials/basic-topics/sorting/insertion-sort.md

```diff
@@ -11,4 +11,204 @@ keywords:
+
+I'll be providing the code in following language:
+
+* C++
+* Java
+* Python
+
+If you don't know any of the above mentioned languages. Don't worry you can easily code insertion sort, in your choice of programming language, once you understand the logic behind it.
+
+## Algorithm
+
+The following are the steps that we are going to follow:-
+
+1. Pick an element from the array, and store it in a variable `nums`.
+2. Now, start comparing the `nums` with all the elements to the left of it.
+3. If the `nums` is less than the current element shift the current element to right until you find an element smaller than `nums`.
+4. Place the `nums` in current position.
+5. Repeat all the above steps until the `array` is sorted.
+
+Let's achieve the sorted `array` by applying above steps in an example.
+
+### Example
+
+Let's apply the insertion sort in an example the You'll get more understanding of the insertion sort algorithm.
+
+We are given with an `array` of height of 5 students (in cm) from a class and we have to sort it.
+
+```python
+height_of_students = [175,180,171,160,168]
+```
+
+1. Let's assume the first element is sorted.Take the second Element store it in `nums`. Now, `175` is smaller than `nums`(`185`) so we'll place `nums` in current position and pick next element. The array would look like this:
+```python
+height_of_students = [175,180,171,160,168]
+```
+2. Now the pick next element assign it to `nums` which in this case is `171`. Now `180` is greater than `nums` so it should be at the right of nums, move `180` to right. Again `175` is greater than `nums` so move `175` to the right. Currently there is no element remain to the left so we will place `nums` in the beginning. The array would look like this:
+```python
+height_of_students = [171,175,180,160,168]
+```
+3. We'll assign the next element to `nums` which is `160` . Till now you have already guessed what is the next step. If your answer is same as step number 3 mentioned above, then you are absolutely correct which is compare the elements to the left of `nums` with it. Now, `180` is greater than `nums`, so move `180` to right. Again `175` is greater than `nums` also move it to the right. Again `171` is smaller than `nums` also move it to right. Now there is no element remains to compare with `nums`, so put `nums` in the beginning of the array. The array would look like this:
+```python
+height_of_students = [160,171,175,180,168]
+```
+4. Now assign `168` to `nums` . And compare with all the elements to the left of it and do the operations according to it. Now, I'm sure you already know what the array would look like:
+```python
+height_of_students = [160,168,171,175,180]
+```
+
+Congrats, You have finally achieved the sorted array with the help of insertion sort algorithm.
+
+## Implementation
+
+Following are the implementation of insertion sort:-
+
+### C++ Implementation
```

Comment:
Use Tabs to show the implementation in different langs.

See `solutions/2400-2499/2418-sort-the-people-easy.md` for reference.